packages/js/src/notifications/helpers.ts

```diff
@@ -23,18 +24,19 @@ export const read = async ({
   try {
     emitter.emit('notification.read.pending', {
       args,
-      optimistic: optimisticValue,
+      data: optimisticValue,
     });
 
     const response = await apiService.read(notificationId);
 
     const updatedNotification = new Notification(response);
-    emitter.emit('notification.read.success', { args, result: updatedNotification });
+    emitter.emit('notification.read.resolved', { args, data: updatedNotification });
 
-    return updatedNotification;
+    return { data: updatedNotification };
   } catch (error) {
-    emitter.emit('notification.read.error', { args, error });
-    throw error;
+    emitter.emit('notification.read.resolved', { args, error });
+
+    return { error: new NovuError('Failed to read notification', error) };
```

Comment:
Custom error objects to provide better error messages. The messages are for the discussion.